backend/endpoints/ratings.py

```diff
@@ -0,0 +1,114 @@
+import os
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from sqlalchemy.sql import func
+from src.database.database import get_db
+from src.database.models import Video, VideoRating, User
+from src.database.schemas import RatingCreate, Rating, VideoWithRatings
+from src.user.auth import get_current_user
+import random
+from typing import Optional
+from pydantic import ValidationError
+
+
+router = APIRouter()
+
+@router.get("/random", response_model=VideoWithRatings)
+async def get_random_video(
+    db: Session = Depends(get_db),
+    current_user: User = Depends(get_current_user)
+):
+    # Get all video IDs that the current user hasn't rated yet
+    rated_video_ids = db.query(VideoRating.video_id).filter(VideoRating.user_id == current_user.id)
+    unrated_videos = db.query(Video).filter(~Video.id.in_(rated_video_ids)).all()
```

Comment:
I believe these 2 queries can be merged to 1, but it isn't important. If it works it works.